Notify admin (#5930)
mattermost/docs: source/upgrade/notify-admin.rst

> Notify admin

Reviewer: @justinegeffen The ``EnableAPITriggerAdminNotifications`` config setting may also need to be documented here:
 - https://docs.mattermost.com/manage/telemetry.html
 - https://docs.mattermost.com/configure/configuration-settings.html

Author: Thank you! I didn't even think of that. :)

Reviewer: Configuration: https://github.com/mattermost/mattermost-server/pull/20777/files#diff-07146319e6e0a169b23444e4db03e086b60c72f16996920a8eabfb057841c8ee

Telemetry: https://github.com/mattermost/mattermost-server/pull/20777/files#diff-7bc66d203ca22000f3fa14a2f1db4d3f013baf12a3b7930f4ad52d1f82b135bb

Author: Thanks so much @amyblais!

============

> In some cases there are Mattermost features that are limited to specific plans. When using Mattermost, users who come across these unavailable features can send a notification to System Admins indicating that they'd like access to this feature.

Reviewer 2: ```suggestion
Some Mattermost features are limited to specific plans. Users who want to access these unavailable features can request access through their System Admin by sending a notification.
```

If access to a feature requires a plan upgrade, System Admins receive notifications of these requests in order to collect data before upgrading. For example, if only one end-user requests access to AD/LDAP, it probably isn't necessary to upgrade. However, if 20 users, including Team and Channel Admins request it, you may want to consider upgrading to support this need.

> This feature is designed for informational purposes only, and no action is required unless you want to take action. Your Mattermost instance's functionality is not affected if you choose not to upgrade.

Reviewer 2: ```suggestion
This feature is designed for informational purposes only, and no action is required unless the System Admin wants to take action. The Mattermost workspace isn't affected if the System Admin chooses not to upgrade.
```

Author: This feature is for System Admins only - end users won't see any of this other than the "notify admin" CTA which has clear explanatory microcopy. Hence "you". What are your thoughts around leaving it as-is given that context?

Notifications
-------------

> Notifications are triggered by users. The very first user trigger results in a bot message to the System Admin, indicating the feature or functionality the user has requested. This bot message is listed in the direct message section of the channel sidebar.

Author: @johndavidlugtu, does this go to all System Admins or just one? If all, what happens if one decides to upgrade and the others don't want to? Is there an upgrade notification sent in that message? If there are multiple admins, who gets the message first?

Reviewer 3: @justinegeffen good questions!

It goes to all System Admins. @AGMETEOR can provide more details for the other questions

Reviewer 3: @justinegeffen 

> If all, what happens if one decides to upgrade and the others don't want to?

One admin upgrading affects the entire workspace. We don't foresee a case where there is a degraded user experience when upgrading.

> Is there an upgrade notification sent in that message?

Once the workspace is upgraded, there is no notification. That said, admins can see in-product multiple places that state their current plan.

> If there are multiple admins, who gets the message first?

They should receive it all simultaneously. Allan can confirm

> Notifications are triggered by users. The very first user trigger results in a bot message to the System Admin, indicating the feature or functionality the user has requested. This bot message is listed in the direct message section of the channel sidebar.

Author: ```suggestion
Notifications are triggered by users. The very first user trigger results in a bot message to all System Admins, indicating the feature or functionality the user has requested. This bot message is listed in the direct message section of the channel sidebar.
```

> Notifications are triggered by users. The very first user trigger results in a bot message to all System Admins, indicating the feature or functionality the user has requested. This bot message is listed in the direct message section of the channel sidebar.

Reviewer 2: ```suggestion
Notifications are triggered by users. The very first user triggers results in a bot message to all System Admins, indicating the feature or functionality the user has requested. This bot message is listed in the **Direct Messages** section of the channel sidebar.
```

Author: This sentence is confusing now that I re-read it. I propose:

"The very first time a user sends a request to upgrade or start a trial, a bot message is sent to all System Admins indicating the feature or functionality the user has requested that requires an upgrade or trial.

Reviewer 2: Thanks! I agree that what's there already is clearer than my attempt to clarify ;)

> Any requests that happen after this first request are sent 14 days later, in a summary format.

Reviewer 3: Propose changing this line to:

Subsequent request notifications are received by admins at most every 14 days. When a notification is received by an admin, a 14-day cool-off period ensues. Any request by end user(s) in the middle of the cool-off period will be held off until 14 days later in a summarized format.

> Any requests that happen after this first request are sent 14 days later, in a summary format.

Author: ```suggestion
Subsequent request notifications are received by admins at most every 14 days. When a notification is received by an admin, a 14-day cool-off period ensues. Any request by end user(s) in the middle of the cool-off period will be held off until 14 days later in a summarized format.
```
